src/Sparrow.Server/ByteString.cs

```diff
@@ -1336,7 +1337,11 @@ public void Release(ref ByteString value)
 
             int reusablePoolIndex = GetPoolIndexForReuse(value._pointer->Size);
 
-            if (value._pointer->Size <= ByteStringContext.MinBlockSizeInBytes)
+            if (value._pointer == _internalCurrent.Current - value._pointer->Size)
+            {
+                _internalCurrent.Current -= value._pointer->Size;
+            }
+            else if (value._pointer->Size <= ByteStringContext.MinBlockSizeInBytes)
             {
```

Comment:
This essentially adds arena allocation mechanism into the `ByteString`, is that correct?

Reply:
Correct.